### PR TITLE
fix(gha): Fix github status log and add tests

### PR DESCRIPTION
**Pull request by jasonmcintosh**
Fixes output logs when using github status checks

### CODE REVIEW

File: echo-notifications/src/test/groovy/com/netflix/spinnaker/echo/config/GithubConfigSpec.groovy [hunk 1]
```diff
@@ -37,6 +43,79 @@ class GithubConfigSpec extends Specification {
     then:
     endpoint.url == ownEndpoint
   }
+
+
+  def 'log level should not default to outputing authorization headers and match basci API call structure'() {
```
Comment (dbyron-sf) [inline echo-notifications/src/test/groovy/com/netflix/spinnaker/echo/config/GithubConfigSpec.groovy:48]:
```suggestion
  def 'default log level does not output authorization headers and matches basic API call structure'() {
```

**Review by dbyron-sf — APPROVED** [review]

**Comment by jasonmcintosh** [comment]
https://github.com/Mergifyio backport 1.30 1.31 1.29 1.28

**Comment by jasonmcintosh** [comment]
https://github.com/Mergifyio backport release-1.30 release-1.31 release-1.29 release-1.28

**Comment by jasonmcintosh** [comment]
https://github.com/Mergifyio backport release-1.30.x release-1.31.x release-1.29.x release-1.28.x

**Pull request merged**